src/main/java/frc/robot/subsystems/RollerIntake.java

```diff
@@ -0,0 +1,125 @@
+
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.Constants;
+import frc.robot.constants.IntakeConstants;
+import frc.robot.util.LogManager;
+import frc.robot.util.MotorFactory;
+
+
+public class RollerIntake extends SubsystemBase {
+
+  public enum IntakeMode {
+    INTAKE_CUBE, OUTTAKE_CUBE, INTAKE_CONE, OUTTAKE_CONE, DISABLED
+  }
+
+  public enum IntakePiece {
+    CUBE, CONE, NONE
+  }
+
+  private final WPI_TalonFX m_intakeMotor;
+  private final ShuffleboardTab m_intakeTab;
+
+  private IntakeMode m_mode;
+  private IntakePiece m_heldPiece;
+
+  private double m_power;
+
+  public RollerIntake(ShuffleboardTab intakeTab) {
+    m_intakeMotor = MotorFactory.createTalonFX(IntakeConstants.kIntakeMotorId, Constants.kRioCAN);
+
+    m_power = 0;
+    m_mode = IntakeMode.DISABLED;
+    // During auto, this doesn't really matter, so we can just set it to NONE
+    // TODO: Auto commands run that pick up a game piece should set this to the correct value
+    m_heldPiece = IntakePiece.NONE;
+    m_intakeTab = intakeTab;
+
+    if (Constants.kUseTelemetry) setupShuffleboard();
+
+  }
+
+  public void setMode(IntakeMode mode){
+    m_mode = mode; 
+  }
+
+  private void setMotorPower(double power){
+    m_intakeMotor.set(power); 
+  }
+
+  @Override
+  public void periodic() {
+
+    switch (m_mode) {
+      case INTAKE_CUBE:
+        m_power = IntakeConstants.kIntakeCubePower;
+        break;
+      case OUTTAKE_CUBE: 
+        m_power = IntakeConstants.kOuttakeCubePower;
+        break;  
+      case INTAKE_CONE: 
+        m_power = IntakeConstants.kIntakeConePower;
+        break;       
+      case OUTTAKE_CONE: 
+        m_power = IntakeConstants.kOuttakeConePower; 
+        break;       
+      case DISABLED: 
+        m_power = 0;
+        break;
+    }
+    // System.out.println(m_mode);
+    setMotorPower(m_power);
+
+    if (Constants.kLogging) {
+      LogManager.addDouble("Intake Motor Current", m_intakeMotor.getStatorCurrent());
+      LogManager.addDouble("Intake Power", m_power);
+    }
+  }
+
+  private void setupShuffleboard() {
+    m_intakeTab.addString("Intake Mode", () -> m_mode.name());
+    m_intakeTab.addDouble("Intake Motor Current", () -> m_intakeMotor.getStatorCurrent());
+    m_intakeTab.addDouble("Intake Power", () -> m_power);
+    m_intakeTab.addString("Held Game Piece", () -> m_heldPiece.name());
+  }
+  
+  public boolean containsGamePiece() {
+    return m_heldPiece != IntakePiece.NONE;
+  }
+
+  public boolean containsCone() {
+    return m_heldPiece == IntakePiece.CONE;
+  }
+
+  public boolean containsCube() {
+    return m_heldPiece == IntakePiece.CUBE;
+  }
+
+  public double getStatorCurrent() {
+    return Math.abs(m_intakeMotor.getStatorCurrent());
```

Comment:
is there a reason you take the absolute value? Also, I think you should use supply current, as that is the current drawn by the motor, not the current provided in the motor controller